### DISCUSSION TITLE
Integer tunes

### DISCUSSION THREAD

**Discussion by elafmusa** [atcollab/at#488] [Python AT]
Hi
The linopt function i use to return the tunes gives only the fractional part, is there a way in PyAT to get the integer part of the tune ?

**Comment by lnadolski** [comment]
Dear @elafmusa,

use atx for instance :
[~, param] = atx(THERING)

param = 
  struct with fields:

              ll: 354.0970
           alpha: 4.1818e-04
       fractunes: [0.1564 0.2283]
       fulltunes: [18.1564 10.2283]
             nuh: 18.1564
             nuv: 10.2283
    chromaticity: [1.3281 2.2883]
     dampingtime: [0.0070 0.0069 0.0035]
        dampingJ: [0.9954 1.0000 2.0046]
         espread: 0.0010
         blength: 0.0046
    modemittance: [3.9347e-09 1.7671e-36 4.7004e-06]
          energy: 2.7391e+09
              fs: 4.3732e+03
           eloss: 9.3459e+05
    synchrophase: 0.3403
      momcompact: 4.1818e-04

Best regards,

Laurent.

**Comment by elafmusa** [comment]
Hi Laurent, thanks, atx is not defines when i use it, is it something i have to add to the modules ?

**Reply by lnadolski** [reply]
You are right this is part of matlab : https://github.com/atcollab/at/blob/master/atmat/atphysics/ParameterSummaryFunctions/atx.m
For python, you can use 

1/ atlinopt and extract the phase advance
   fulltune = LinData(end).mu/2/pi

2/ get_tune with extra flag
initial_tune = get_tune(ring, ..., use_integer = True)
